### ISSUE TITLE
Pluggable validation

### ISSUE THREAD

**Issue by aldeed** [meteor/validated-method#4]
Rather than depending on simple-schema, maybe I can add a function that throws a ValidationError in the simple-schema pkg.

```
validation: new SimpleSchema({
    listId: { type: String }
  }).validator()
```

And if you also want to keep the nice `schema` option, maybe by convention this pkg could call a `validator` function property on whatever is passed to `schema` without caring what it is. (`validator` would return a function that validates and throws ValidationErrors.)

**Comment by stubailo** [comment]
This is a great idea!

**Comment by tmeasday** [comment]
@aldeed something else very close to this that would be super useful would be:

``` js
// throw a ValidationError if it fails
new SimpleSchema({
  listId: { type: String } 
}).validate(doc);
```

(which would of course make `.validator()` trivial to implement).

What do you think? Can I help make stuff happen or is it easier to leave it to you and your refactor?

**Comment by stubailo** [comment]
@tmeasday I think the simplest way would be to fuse the two APIs:

``` js
validate: new SimpleSchema({
  listId: { type: String }
}).validate
```

As long as the `validate` function is bound to the simple-schema instance, this should enable both APIs at once.

**Comment by aldeed** [comment]
@stubailo @tmeasday My thinking with `.validator()` rather than `.validate` is that having a function return a function would keep the door open for passing in options that people might want in the future (like ignore X type of errors, custom error message, etc.) Also, yes, I am adding `.validate` as part of the v2 refactor I'm working on, and the function returned by `.validator()` would call that. I probably have a few weeks left before I'm done with that, though, so I can patch v1 in the meantime.

**Comment by tmeasday** [comment]
@aldeed it would be really awesome if you could patch v1, as it would mean I'm not lying here: http://meteor-guide.s3-website-us-west-1.amazonaws.com/collections.html#Checking_documents_against_a_schema

But I understand if you want to wait for v2 also, the guide won't be official for more than a few weeks anyway.

**Comment by aldeed** [comment]
@stubailo @tmeasday How does this look? https://github.com/aldeed/meteor-simple-schema/commit/cfc075c6dbd87c31eb494cb4fce4b593cd2f8313

I can publish as 1.4 if we're on the same page.

**Comment by tmeasday** [comment]
LGTM @stubailo?

**Comment by stubailo** [comment]
Amazing!

**Comment by aldeed** [comment]
Cool. Published.

**Comment by stubailo** [comment]
OK going to try to get to all of these issues this week and publish a new version of this package.

**Comment by stubailo** [comment]
Looks like this is fixed as of #7
